service/waterfall.go

```diff
@@ -395,6 +397,58 @@ func getVersionsAndVariants(skip, numVersionElements int, project *model.Project
 
 }
 
+func (restapi restAPI) getVersionsAndVariantsApi(w http.ResponseWriter, r *http.Request) {
```

Comment:
This should be a RestV2 route and also should be named something like "getRecentVersions" rather than referencing the waterfall

Reply:
@john-m-liu , Can you please give me an example of v2 route? And also why v2?
And yes, it would be really nice if the code would be merged asap, if there are no any major issues.

Reply:
They are the routes in rest/route/service.go. They use a more robust API model than the ones in service/

Reply:
Got it. I'll redo it this week.